src/cryptoadvance/specter/helpers.py

```diff
@@ -260,29 +276,33 @@ def parse_wallet_data_import(wallet_data):
             xpubs += "[{}]{}/0/*,".format(
                 d["derivation"].replace("m", d["root_fingerprint"]), d["xpub"]
             )
-            cosigners_types.append(d["hw_type"])
+            cosigners_types.append({"type": d["hw_type"], "label": d["label"]})
             i += 1
         xpubs = xpubs.rstrip(",")
-        if wallet_data["addresses"]["receiving"][0].startswith("bc") or wallet_data[
-            "addresses"
-        ]["receiving"][0].startswith("tb"):
-            wallet_type = "wsh"
+
+        if "derivation" in wallet_data["x1/"]:
```

Comment:
I just checked with Electrum file format, looks like the most reliable way to detect script type is by looking at `xpub` field in the keys - they strictly use slip132:
- `wsh` uses `Zpub` (`Vpub` on testnet)
- `sh-wsh` uses `Ypub` (`Upub` on testnet)
- `sh` uses `xpub` (`tpub` on testnet).

For single-sig it's `zpub / vpub` for `wpkh`, `ypub / upub` for `sh-wpkh` and `xpub / tpub` for `pkh`.

Derivation is not a very reliable source of information as the seed generated on electrum itself has derivation `m/1'` (at least for `wsh`). It is also possible to import xpubs directly to Electrum without any derivation paths.

So I think the best way to properly parse electrum wallet type is to check prefix of the `xpub` field in the dictionary.

Reply:
Awesome. That's much more straightforward. Update coming.